gatsby/content/docs/e2ee_cross_signing.mdx

```diff
@@ -110,6 +117,9 @@ set to the ID of the sending device. So the above payload becomes:
 
 For simplicity, all payloads mentioned below will be shown without those extra keys.
 
+While the to_device messages are sent encrypted, the room messages don't *have* to be encrypted - it
```

Comment:
The to_device messages are not sent encrypted. They are always sent in the clear.

Reply:
Huh, riot at least works if you send them encrypted. So maybe change that they *can* be encrypted, then?

Reply:
Clients will usually parse the decrypted event the same way it would parse an unencrypted one (except that they will remember that the event was encrypted).

So they will mostly accept encrypted versions as well. Riot doesn't send them encrypted and I don't think any of the mobile Riots do so either. The spec doesn't mention anywhere anything about them being encrypted either, though it doesn't mention that they can't be either.

There is not much to gain by sending those events encrypted, the server won't know that you're currently verifying the device that should receive the message.

That being said there is not much downside either, it's a bit more complex since you need to setup Olm a session before you can start verifying a device.

Reply:
> That being said there is not much downside either, it's a bit more complex since you need to setup Olm a session before you can start verifying a device.

Ideally you'll have a method that sends a to_device method that does all that stuff for you anyways, so for the client developer it'd be zero-effort to add encryption. That being said, soru will update that section accordingly.

Reply:
By more complex I didn't mean that a developer needs to do this, I mean that a client needs to go through a more complex path before it can do the verification dance which might introduce more delays that end up being noticeable to the user.

It's not that important since the same problem can happen with in-room verifications.

Reply:
Alright, adjusted the paragraph!